conda/cli/activate.py

```diff
@@ -24,95 +38,105 @@ def help():
     return prefix
 
 
 def binpath_from_arg(arg):
-    path = join(prefix_from_arg(arg), 'bin')
-    if not isdir(path):
-        sys.exit("Error: no such directory: %s" % path)
+    prefix = prefix_from_arg(arg)
+    if on_win:
+        path = [prefix.rstrip("\\"),
+                join(prefix, 'Scripts'),
+                join(prefix, 'Library', 'bin'),
+               ]
+    else:
+        path = [prefix.rstrip("/"),
+                join(prefix, 'bin'),
+                ]
     return path
 
+
+def pathlist_to_str(paths):
+    """
+    Format a path list, e.g., of bin paths to be added or removed,
+    for user-friendly output.
+    """
+    return ' and '.join([path.replace("\\\\", "\\") for path in paths])
+
+
 def main():
+    import conda.config
+    import conda.install
     if '-h' in sys.argv or '--help' in sys.argv:
         help()
 
     if sys.argv[1] == '..activate':
         if len(sys.argv) == 2:
-            sys.exit("Error: no environment provided.")
+            binpath = binpath_from_arg("root")
         elif len(sys.argv) == 3:
             binpath = binpath_from_arg(sys.argv[2])
         else:
             sys.exit("Error: did not expect more than one argument")
-
-        paths = [binpath]
-        sys.stderr.write("prepending %s to PATH\n" % binpath)
+        sys.stderr.write("prepending %s to PATH\n" % pathlist_to_str(binpath))
+        path = os.pathsep.join([os.pathsep.join(binpath), os.getenv("PATH")])
 
     elif sys.argv[1] == '..deactivate':
-        if len(sys.argv) != 2:
-            sys.exit("Error: too many arguments.")
-
-        try:
-            binpath = binpath_from_arg(os.getenv('CONDA_DEFAULT_ENV', 'root'))
-        except SystemExit:
-            print(os.environ['PATH'])
-            raise
-        paths = []
-        sys.stderr.write("discarding %s from PATH\n" % binpath)
-
-    elif sys.argv[1] == '..activateroot':
-        if len(sys.argv) != 2:
-            sys.exit("Error: too many arguments.")
-
-        if 'CONDA_DEFAULT_ENV' not in os.environ:
-            sys.exit("Error: No environment to deactivate")
-        try:
-            import conda.config
-            binpath = binpath_from_arg(os.getenv('CONDA_DEFAULT_ENV'))
-            rootpath = binpath_from_arg(conda.config.root_env_name)
-        except SystemExit:
-            print(os.environ['PATH'])
-            raise
-        # deactivate is the same as activate root (except without setting
-        # CONDA_DEFAULT_ENV or PS1). XXX: The user might want to put the root
-        # env back somewhere in the middle of the PATH, not at the beginning.
-        if rootpath not in os.getenv('PATH').split(os.pathsep):
-            paths = [rootpath]
-        else:
-            paths = []
-        sys.stderr.write("discarding %s from PATH\n" % binpath)
+        path = os.getenv("PATH")
+        if os.getenv("CONDA_ACTIVE_ENV"):
+            binpath = binpath_from_arg(os.getenv('CONDA_ACTIVE_ENV'))
```

Comment:
I think in those two lines, the os.getenv should have a default of `'root'`. When starting with a new shell, the root environment is implicitly activate, so when activating another env, the root env should be discarded first.

I would suggest to change it into `os.getenv('CONDA_ACTIVE_ENV', 'root')`.